packages/ui/src/components/Checkbox/checkbox.tsx

```diff
@@ -10,23 +11,35 @@ type CondoCheckboxProps = {
     labelProps?: TypographyTextProps
 }
 
-export type CheckboxProps = Pick<DefaultCheckboxProps, 'autoFocus' | 'defaultChecked' | 'disabled' | 'onChange' | 'indeterminate' | 'checked'> & CondoCheckboxProps
+export type CheckboxProps = Pick<DefaultCheckboxProps, 'autoFocus' | 'defaultChecked' | 'disabled' | 'onChange' | 'indeterminate' | 'checked' | 'children' | 'id'> & CondoCheckboxProps
 
 const Checkbox: React.FC<CheckboxProps> = (props) => {
-    const { label, labelProps, disabled, ...rest } = props
+    const { label, labelProps, disabled, id, onChange, children, ...rest } = props
+
+    const handleChange = useCallback((event) => {
+        const stringContent = label ? label : extractChildrenContent(children)
+        if (stringContent) {
+            sendAnalyticsCheckEvent('Checkbox', { value: stringContent, id })
+        }
+
+        if (onChange) {
```

Comment:
if (isFunction(onChange))

Reply:
If we add this check, then we will not report that something went wrong and the type is incorrect. If not a function was passed, then an error should fall - by default this happens